src/Admin/Settings.php

```diff
@@ -123,6 +123,49 @@ function () {
 					]
 				);
 
+				register_graphql_settings_field(
+					'graphql_persisted_queries_section',
+					[
+						'name'              => 'query_gc',
+						'label'             => __( 'Clean Up Queries', 'wp-graphql-smart-cache' ),
+						'desc'              => __( 'Toggle on to enable garbage collection of saved queries older than number of days specified below', 'wp-graphql-smart-cache' ),
+						'type'              => 'checkbox',
+						'default'           => 'off',
+						'sanitize_callback' => function ( $value ) {
+							/**
+							 * When enable garbage collection,
+							 * schedule the garbage collection action/event to run once daily.
+							 * Otherwise remove it.
+							 */
+							if ( 'on' === $value ) {
+								if ( ! wp_next_scheduled( 'wp_graphql_smart_cache_query_gc' ) ) {
+									// Add scheduled job to run in one minute
+									wp_schedule_event( time() + 60, 'daily', 'wp_graphql_smart_cache_query_gc' );
```

Comment:
Should we run this more than daily?

Reply:
ah ok. for some reason I thought the `xx days ago` query was based on a day, not a timestamp. 

I just confirmed that:

```php
'date_query' => [
    [
        'column' => 'post_modified_gmt',
        'before' => $days_ago . ' days ago',
    ],
]
```

does indeed translate to:

```php
SELECT SQL_CALC_FOUND_ROWS  wp_posts.ID FROM wp_posts WHERE 1=1  AND ( wp_posts.post_modified_gmt < '2023-07-19 17:40:05') AND ((wp_posts.post_type = 'post' AND (wp_posts.post_status = 'publish'))) ORDER BY wp_posts.post_date DESC, wp_posts.ID DESC LIMIT 0, 10 
```

With the key being: `'2023-07-19 17:40:05'`

I thought it might have simply been `2023-07-19` or similar.

Reply:
So back to the original question:

> Should we run this more than daily?

I think we can start with 1x/daily, but we should have a filter that allows users to override that. 

Perhaps apply a filter over the `daily` value and users can override that value if needed. That might be nice for our own testing sake even, where we could drop a snippet in that filters that to run more often than daily so we can debug things locally easier, perhaps?